main.c

```
@@ -570,14 +602,35 @@ struct process_config *parse_process_config(char **string_area, size_t max_sz) {
 				fprintf(stderr, "Failed to retreive WD information from %s\n", tmp_field);
 				break;
 			}
+		} else if (memcmp(tmp_field, "RLIMIT", 6) == 0) {
+            struct urunit_rlimit tmp_rlim;
+            // Parse the line using the helper function
+            ret = get_rlimit_val(tmp_field, &tmp_rlim);
+            if (ret != 0) {
+                fprintf(stderr, "Failed to retrieve RLIMIT information from %s\n", tmp_field);
+                break;
+            }
+            
+            // Expand the array to hold the new limit
+            conf->rlimits_count++;
+            struct urunit_rlimit *new_ptr = realloc(conf->rlimits, conf->rlimits_count * sizeof(struct urunit_rlimit));
+            if (!new_ptr) {
+                fprintf(stderr, "Failed to allocate memory for rlimits\n");
+                // Don't leak the previous pointer if realloc failed (though typically we exit here)
+                break;
```

If memory allocation fails, we need to return an error and exit.